File: backend/core-services/application-service/src/main/java/com/govtech/application/domain/model/Application.java
package com.govtech.application.domain.model;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

import lombok.Builder;

@Builder
public record Application(
                UUID applicationId,
                String subject,
                String aidCode,
                String aidName,
                ApplicationStatus status,
                String objectKey,
                Instant createdAt,
                List<DocumentType> missingDocuments) {
}

File: frontend/portal-web/src/features/application/types/application.ts
import { DocumentType } from "./document-type";


export interface Application {
    id: string;
    aidName: string;
    aidCode: string;
    status:
    | "GENERATED"
    | "READY_TO_COMPLETE"
    | "READY_TO_SUBMIT"
    | "SUBMITTED"
    | "ACCEPTED"
    | "REJECTED";
    createdAt: string;
    missingDocuments: DocumentType[];
}